### PR TITLE
Романов - Лабораторная работа #2

### PR DESCRIPTION
**Pull request by xnd-r**

### CODE REVIEW

**Comment by xnd-r** [comment]
@katerina394 @VladimirNechesanov please review

**Comment by katerina394** [comment]
👍

**Comment by xnd-r** [comment]
@VlaKuz could u please review

File: code/krasikova_ekaterina_lab1/view/dheap_view.py [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+    def __init__(self):
+        self.root = tk.Tk()
+        self.root.title('D-Heap')
+        self.root.geometry('900x200')
+        self.root.resizable(width=False, height=False)
+
+        self.frame = tk.Frame(self.root)
+
+        self.d_label = tk.Label(self.root, text="D:")
+        self.d = tk.Entry(self.root, width=10)
```
Comment (VlaKuz) [inline code/krasikova_ekaterina_lab1/view/dheap_view.py:19]: Please rename it to self.entry_param_d for example. This is necessary to better understand the meaning of the variable.

File: code/krasikova_ekaterina_lab1/viewmodel/dheap_viewmodel.py [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+            try:
+                self.deleting_elem = int(elem)
+                self.set_delete_btn_enabled()
+            except:
+                self.set_delete_btn_disabled()
+
+    def set_decreasing_elem(self, elem):
+        if self.is_heap_init:
+            try:
+                decrease_data = [float(i) for i in elem.split(' ')]
```
Comment (VlaKuz) [inline code/krasikova_ekaterina_lab1/viewmodel/dheap_viewmodel.py:70]: same

File: code/krasikova_ekaterina_lab1/viewmodel/dheap_viewmodel.py [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+                self.decrease_value = float(decrease_data[1])
+                self.set_decrease_btn_enabled()
+            except:
+                self.set_decrease_btn_disabled()
+
+    def set_input_data(self, d, input_data):
+        self.is_d = False
+        self.is_input_data = False
+        try:
+            input_data = [float(i) for i in input_data.split(' ')]
```
Comment (VlaKuz) [inline code/krasikova_ekaterina_lab1/viewmodel/dheap_viewmodel.py:81]: same

File: code/krasikova_ekaterina_lab1/viewmodel/dheap_viewmodel.py [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+            else:
+                self.set_create_btn_disabled()
+        except:
+                self.set_create_btn_disabled()
+        if all([self.is_input_data, self.is_d]):
+            self.set_create_btn_enabled()
+
+    def create_btn_clicked(self):
+        self.DHeap = DHeap(self.d, self.input_data)
+        self.current_data = ' '.join([str(i) for i in self.DHeap.heap])
```
Comment (VlaKuz) [inline code/krasikova_ekaterina_lab1/viewmodel/dheap_viewmodel.py:99]: same

File: code/krasikova_ekaterina_lab1/viewmodel/dheap_viewmodel.py [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+        self.is_heap_init = True
+
+    def insert_btn_clicked(self):
+        if self.is_heap_init:
+            self.DHeap.insert(self.inserting_elem)
+            self.current_data = ' '.join([str(i) for i in self.DHeap.heap])
```
Comment (VlaKuz) [inline code/krasikova_ekaterina_lab1/viewmodel/dheap_viewmodel.py:105]: same

File: code/krasikova_ekaterina_lab1/viewmodel/dheap_viewmodel.py [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+
+    def delete_btn_clicked(self):
+        self.DHeap.delete(self.deleting_elem)
+        self.current_data = ' '.join([str(i) for i in self.DHeap.heap])
```
Comment (VlaKuz) [inline code/krasikova_ekaterina_lab1/viewmodel/dheap_viewmodel.py:109]: same

File: code/krasikova_ekaterina_lab1/viewmodel/dheap_viewmodel.py [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+
+    def decrease_btn_clicked(self):
+        self.DHeap.decrease_weight(self.decrease_key, self.decrease_value)
+        self.current_data = ' '.join([str(i) for i in self.DHeap.heap])
```
Comment (VlaKuz) [inline code/krasikova_ekaterina_lab1/viewmodel/dheap_viewmodel.py:113]: same

**Comment by VlaKuz** [comment]
👍

**Pull request merged**